osu.Game/Beatmaps/ControlPoints/ControlPointInfo.cs

```diff
@@ -12,6 +12,11 @@ namespace osu.Game.Beatmaps.ControlPoints
     [Serializable]
     public class ControlPointInfo
     {
+        private static readonly TimingControlPoint default_timing_point = new TimingControlPoint();

```

Comment:
let's hope no one changes these by accident >.>

Reply:
Yeah I considered that too, these should definitely be readonly or structs.

Reply:
is this just a preoptimisation? or is there another reason for it?

Reply:
Could attempt but not sure how big of a change

Reply:
I mean what's the reason for static in the first place?

Reply:
Actually you bring up a good point, I could override GetHashCode instead... It's because these are used for dictionary keys.